From f306335d0f7cbe362cc6dd869642b65376db8b6c Mon Sep 17 00:00:00 2001
From: Niall Pemberton <niallp@apache.org>
Date: Fri, 15 Feb 2008 20:14:20 +0000
Subject: [PATCH] Fix for BEANUTILS-306 - LocaleConvertUtilsBean.convert throws
 NPE on null Locale when debug logging is enabled (NOTE: change surefire
 plugin configuration logging level to "DEBUG" in pom.xml to test this is
 fixed)

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/beanutils/trunk@628158 13f79535-47bb-0310-9956-ffa450edef68
---
 pom.xml                                       | 17 ++++++++++
 .../locale/LocaleConvertUtilsBean.java        |  4 +--
 .../locale/LocaleConvertUtilsTestCase.java    | 34 +++++++++++++++++++
 3 files changed, 53 insertions(+), 2 deletions(-)

diff --git a/pom.xml b/pom.xml
index e5595666c..f37e242ac 100644
--- a/pom.xml
+++ b/pom.xml
@@ -180,6 +180,23 @@
                 -->
                 <exclude>**/*MemoryTestCase.java</exclude>
               </excludes>
+
+              <!-- Configure Logging -->
+              <systemProperties>
+                  <property>
+                      <name>org.apache.commons.logging.LogFactory</name>
+                      <value>org.apache.commons.logging.impl.LogFactoryImpl</value>
+                  </property>
+                  <property>
+                      <name>org.apache.commons.logging.Log</name>
+                      <value>org.apache.commons.logging.impl.SimpleLog</value>
+                  </property>
+                  <property>
+                      <name>org.apache.commons.logging.simplelog.defaultlog</name>
+                      <value>WARN</value>
+                  </property>
+              </systemProperties>
+
           </configuration>
         </plugin>
         <plugin>
diff --git a/src/java/org/apache/commons/beanutils/locale/LocaleConvertUtilsBean.java b/src/java/org/apache/commons/beanutils/locale/LocaleConvertUtilsBean.java
index 70a5f701d..d6896f453 100644
--- a/src/java/org/apache/commons/beanutils/locale/LocaleConvertUtilsBean.java
+++ b/src/java/org/apache/commons/beanutils/locale/LocaleConvertUtilsBean.java
@@ -263,7 +263,7 @@ public Object convert(String value, Class clazz, Locale locale, String pattern)
 
         if (log.isDebugEnabled()) {
             log.debug("Convert string " + value + " to class " +
-                    clazz.getName() + " using " + locale.toString() +
+                    clazz.getName() + " using " + locale +
                     " locale and " + pattern + " pattern");
         }
 
@@ -333,7 +333,7 @@ public Object convert(String[] values, Class clazz, Locale locale, String patter
         }
         if (log.isDebugEnabled()) {
             log.debug("Convert String[" + values.length + "] to class " +
-                    type.getName() + "[] using " + locale.toString() +
+                    type.getName() + "[] using " + locale +
                     " locale and " + pattern + " pattern");
         }
 
diff --git a/src/test/org/apache/commons/beanutils/locale/LocaleConvertUtilsTestCase.java b/src/test/org/apache/commons/beanutils/locale/LocaleConvertUtilsTestCase.java
index 911d7fa36..df70c7253 100644
--- a/src/test/org/apache/commons/beanutils/locale/LocaleConvertUtilsTestCase.java
+++ b/src/test/org/apache/commons/beanutils/locale/LocaleConvertUtilsTestCase.java
@@ -23,6 +23,7 @@
 import java.sql.Timestamp;
 import java.text.DecimalFormat;
 import java.text.NumberFormat;
+import java.util.Locale;
 
 import junit.framework.TestCase;
 import junit.framework.Test;
@@ -595,6 +596,39 @@ public void fixmetestPositiveStringArray() {
 
     }
 
+    /**
+     * Test conversion of a String using a Locale and pattern.
+     */
+    public void testConvertStringLocaleNull() {
+        Object result = null;
+        try {
+            result = LocaleConvertUtils.convert("123", Integer.class, (Locale)null, "#,###");
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Threw: " + e);
+        }
+        assertNotNull("Null Result", result);
+        assertEquals("Integer Type", Integer.class, result.getClass());
+        assertEquals("Integer Value", new Integer(123), result);
+    }
+
+    /**
+     * Test conversion of a String array using a Locale and pattern.
+     */
+    public void testConvertStringArrayLocaleNull() {
+        Object result = null;
+        try {
+            result = LocaleConvertUtils.convert(new String[] {"123"}, Integer[].class, (Locale)null, "#,###");
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Threw: " + e);
+        }
+        assertNotNull("Null Result", result);
+        assertEquals("Integer Array Type", Integer[].class, result.getClass());
+        assertEquals("Integer Array Length", 1, ((Integer[])result).length);
+        assertEquals("Integer Array Value", new Integer(123), ((Integer[])result)[0]);
+    }
+
 
     // -------------------------------------------------------- Private Methods
 
